utils/urls.js

```diff
@@ -0,0 +1,27 @@
+//fetch Vercel baseurl based on current status
+function getBaseURL () {
+    // if Deployement
+    if (process.env.VERCEL_URL) {
+      return `https://${process.env.VERCEL_URL}`
+    }
+    // if Build Step
+    if (process.env.NEXT_PUBLIC_VERCEL_URL) {
+      return `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`
+    }
+    return 'http://localhost:3000'
+  }
```

Comment:
is this ever required-- present or future? lmk what your plans are for this

Reply:
Here is my idea:
next js routing separated client-side pages and server-side prompt, in order to fetch data at the client-side we'll need to go through various api endpoints as the example shows https://nextjs.org/docs/basic-features/data-fetching/client-side, we must call fetch on api endpoint, and to my understanding fetch only supports absolute URL(`../../api/test` is not supported), this absolute url have the format like baseurl+api route, take an example if I want to fetch data from test api login page I must call:
`fetch("http://localhost:3000/api/test")`
however, if we were at vercel this url will be changed into something like this:
`fetch("https://boggers-qzvf4fwx9-samarth52.vercel.app/api/test")` 
Note that this url is different in each deployment, hence it's impossible for us to hardcode the api url we need to fetch, and we will need to fetch data at different pages from the index, to that end I wrote these lines, it will test if the current environment is vercel(these environmental variables are automatically set by vercel) if it does it will return the Vercel base url. If not it return the default local url.